.github/workflows/push-selfcontained.yml

```
@@ -0,0 +1,90 @@
+name: Push self-contained images on Dockerfile change
+
+on:
+  push:
+    branches:
+    - main
```

```suggestion
    - main
    paths:
    - 'images/test-environments/**/Dockerfile`
```

I think you can filter this to run only on those pushes that update the self-contained images: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows?ref=hackernoon.com&msclkid=e30890edcb2411ecb18b3cc68805c075#running-your-workflow-only-when-a-push-affects-specific-files